static-metric/proc-macros/src/auto_flush_from.rs

```diff
@@ -0,0 +1,43 @@
+use proc_macro2::Span;
+use syn::export::TokenStream2;
+use syn::parse::{Parse, ParseStream};
+use syn::token::*;
+use syn::*;
+
+pub struct AutoFlushFromDef {
+    class_name: Ident,
+    inner_class_name: Ident,
+    source_var_name: Ident,
```

Comment:
I suggest to use `Expr` rather than `Ident`. Then the user can also pass in an expression.